pkg/fetcher/fetcher.go

```diff
@@ -64,17 +64,20 @@ func fetchUpstreamHelmrepo(upstreamYaml upstreamyaml.UpstreamYaml) (ChartSourceM
 
 	resp, err := http.Get(url)
 	if err != nil {
-		return chartSourceMeta, err
+		return chartSourceMeta, fmt.Errorf("request to %s failed: %w", url, err)
+	}
+	if resp.StatusCode >= 300 {
```

Comment:
Does every 2xx code work? Even something like 204 No Content?

Reply:
I guess this makes it work. I'm not sure what to do if we have to worry about specific 2xx codes. FWIW, this is a `GET` request, so I don't think we'll ever get a 204. I'd say that the chance of getting a 2xx that is not a 200 is quite low.

The general idea is to catch the cases where the server has been moved, or taken down. Even capturing 4xx and 5xx codes is probably sufficient for this, given that `http.Get()` follows a limited number of redirects.

How would you suggest we proceed? I'm open to ideas :smile:

Reply:
I think adding a log when we get an error code that is not 200 would suffice. If it happens a lot, then we can think of how to fix it